### PR TITLE
Clean up extractCritical code

### PR DESCRIPTION
**Pull request by mattfysh**
**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Code complete

### CODE REVIEW

File: packages/emotion-server/src/index.js [hunk 1]
```diff
@@ -15,12 +15,10 @@ export function extractCritical(html) {
...
   }
-
-  o.rules = sheet.sheet.slice().filter(x => {
-    RGX.lastIndex = 0
-    let match = RGX.exec(x)
-    const ret = match == null || ids[match[1]] || false
-    return ret
+  
+  o.rules = sheet.sheet.filter(x => {
+    const match = x.match(RGX)
```
Comment (tkh44) [inline packages/emotion-server/src/index.js]: Are you sure we want to remove the `RGX.lastIndex = 0` line? This is a global regex and the index needs to reset after every run from my understanding.

Comment (mattfysh) [reply packages/emotion-server/src/index.js]: that's right for the way it was used before `RGX.exec(x)`, however it's not applied when using `String.prototype.match`

Comment (tkh44) [reply packages/emotion-server/src/index.js]: 👍 missed that part 😅

File: packages/emotion-server/src/index.js [hunk 2]
```diff
@@ -15,12 +15,10 @@ export function extractCritical(html) {
...
+    return !match || ids[match[0]]
```
Comment (mitchellhamilton) [inline packages/emotion-server/src/index.js]: Could you change `!match` to `match === null`, it's a bit more explicit and there will be a (extremely small and probably unimportant) performance improvement.

**Comment by mattfysh** [comment]
sorry all, I'm getting slammed with work and won't have time to update. As this is purely cosmetic feel free to close the PR

**Pull request closed without merging**